### PR TITLE
Remove env vars and commandline when the request is coming from ARM.

### PR DESCRIPTION
**Pull request by ahmelsayed**

### CODE REVIEW

File: Kudu.Services/Diagnostics/ProcessController.cs [hunk 1]
```diff
@@ -456,6 +456,12 @@ private ProcessInfo GetProcessInfo(Process process, string href, bool details =
                 SetEnvironmentInfo(info);
             }
 
+            if (ArmUtils.IsArmRequest(Request))
+            {
+                info.EnvironmentVariables = null;
+                info.CommandLine = null;
+            }
```
Comment (ahmelsayed) [inline Kudu.Services/Diagnostics/ProcessController.cs]: we could have populating these as a conditional above, but then we will lose the info for 

```csharp
processInfo.IsScmSite 
processInfo.IsWebJob 
processInfo.Description
```

which are deduced from the environment variables.

**Comment by ahmelsayed** [comment]
The only other `GET` that could be problematic is the one for `/dump`, but I don't think would work over ARM, right @suwatch?

File: Kudu.Services/Diagnostics/ProcessController.cs [hunk 2]
```diff
@@ -456,6 +456,12 @@ private ProcessInfo GetProcessInfo(Process process, string href, bool details =
                 SetEnvironmentInfo(info);
             }
 
+            if (ArmUtils.IsArmRequest(Request))
```
Comment (davidebbo) [inline Kudu.Services/Diagnostics/ProcessController.cs]: Why not just `if (!ArmUtils.IsArmRequest(Request))` in the code above where you set them in the first place?

Comment (ahmelsayed) [reply Kudu.Services/Diagnostics/ProcessController.cs]: I just mentioned that here https://github.com/projectkudu/kudu/pull/2868#discussion_r219310329 
 
> we could have populating these as a conditional above, but then we will lose the info for
```
processInfo.IsScmSite 
processInfo.IsWebJob 
processInfo.Description
```
>which are deduced from the environment variables.

**Comment by davidebbo** [comment]
@ahmelsayed  right, no way dump would work over ARM!

**Pull request merged**